bot/exts/events/trivianight/_scoreboard.py

```diff
@@ -0,0 +1,128 @@
+            if (current_placement + 1) % 10 == 0:
+                formatted_string += "⎯⎯⎯⎯⎯⎯⎯⎯\n"
+
+        speed_embed = Embed(
+            title="Average time taken to answer a question",
+            description=formatted_string,
+            color=Colours.python_blue
+        )
+        return speed_embed
+
+    def _get_rank(self, member: Member) -> Embed:
+        """Gets the member's rank for the points leaderboard and speed leaderboard."""
+        rank_embed = Embed(title=f"Ranks for {member.display_name}", color=Colours.python_blue)
+        # These are stored as strings so that the last digit can be determined to choose the suffix
+        try:
+            points_rank = str(list(self.points.keys()).index(member.id) + 1)
+            speed_rank = str(list(self.speed.keys()).index(member.id) + 1)
+        except ValueError:
+            return Embed(
+                title=choice(NEGATIVE_REPLIES),
+                description="It looks like you didn't participate in the Trivia Night event!",
+                color=Colours.soft_red
+            )
+
+        suffixes = {"1": "st", "2": "nd", "3": "rd"}
+        rank_embed.add_field(
+            name="Total Points",
+            value=(
+                f"You got {points_rank}{'th' if not (suffix := suffixes.get(points_rank[-1])) else suffix} place"
+                f" with {self.points[member.id]:.1f} points."
+            ),
+            inline=False
+        )
+        rank_embed.add_field(
+            name="Average Speed",
+            value=(
+                f"You got {speed_rank}{'th' if not (suffix := suffixes.get(speed_rank[-1])) else suffix} place"
+                f" with a time of {(self.speed[member.id][1] / self.speed[member.id][0]):.1f} seconds."
+            ),
+            inline=False
+        )
+        return rank_embed
+
+    @discord.ui.button(label="Scoreboard for Speed", style=ButtonStyle.green)
+    async def speed_leaderboard(self, button: Button, interaction: Interaction) -> None:
+        """Send an ephemeral message with the speed leaderboard embed."""
+        await interaction.response.send_message(embed=await self._create_speed_embed(), ephemeral=True)
+
+    @discord.ui.button(label="What's my rank?", style=ButtonStyle.blurple)
+    async def rank_button(self, button: Button, interaction: Interaction) -> None:
+        """Send an ephemeral message with the user's rank for the overall points/average speed."""
+        await interaction.response.send_message(embed=self._get_rank(interaction.user), ephemeral=True)
+
+
+class Scoreboard:
+    """Class for the scoreboard for the Trivia Night event."""
+
+    def __setitem__(self, key: str, value: int):
+        if value.get("points") and key.user_id not in self.view.points.keys():
+            self.view.points[key.user_id] = value["points"]
+        elif value.get("points"):
+            self.view.points[key.user_id] += self.view.points[key.user_id]
```

Comment:
```suggestion
        if value.get("points"):
            self.view.points[key.user_id] += self.view.points.get(key.user_id, value["points"])
```
If you accept this repeat the same for speed.

Reply:
Wouldn't this error out if `key.user_id` isn't a key inside `self.view.points`?

Reply:
That's what the `.get()` is, the second argument is the default when that key isn't already present.

Reply:
Oh nvm - I just realized - that does `+=` vs. `=` for when it is missing.

Yeah you're right here.